### PR TITLE
Terminate links on Narrow Non-Breaking Spaces

### PR DESCRIPTION
**Pull request by ckumar1**
Creates an isUnicodeSpace method that checks for Narrow Non Breaking Spaces in addition to the usual ASCII whitespace.

/cc @zendesk/sustaining @pdeuter
### References
- Ticket link: https://support.zendesk.com/agent/tickets/1221925
### Risks
- Low: Links in tickets are rendered incorrectly

### CODE REVIEW

**Comment by pdeuter** [comment]
+1  @jcheatham

**Comment by jcheatham** [comment]
Looks good, but with a name like that I wonder if it shouldn't be expanded to include all of the other unicode spaces too.  In which case you may want to implement a loose state table approach, or something to take into account that several of them start with the same prefix.

**Comment by Nebopolis** [comment]
I wonder if it would be possible to generalize this by using an array of whitespace characters and `strncmp(data[offset], whitespace[index], sizeof(whitespace[index])`? Didn't think of this when we were pairing, but seems like something that would be good to investigate.

**Comment by ckumar1** [comment]
@jcheatham @Nebopolis I added in the remaining unicode spaces and got it to iterate through correctly

**Comment by pdeuter** [comment]
These spaces are encoded as UTF-16 ... I guess that is correct?  Normally, I think our characters are encoded as UTF-8.

File: ext/rinku/autolink.c [hunk 1]
```diff
@@ -313,3 +313,40 @@ sd_autolink__url(
 
 	return link_end;
 }
+
+int
+isUnicodeSpace(uint8_t *data, size_t offset) {
```
Comment (bquorning) [inline ext/rinku/autolink.c]: Reminds me of https://github.com/SamSaffron/fast_blank/blob/master/ext/fast_blank/fast_blank.c

**Comment by Nebopolis** [comment]
It would be nice to have some tests for various unicode spaces? @pdeuter also makes a good point about UTF-16 vs UTF-8. I'm not sure if that will work.

**Comment by pdeuter** [comment]
+1 I think this is probably fine wrt encoding ... the test contains a nbsp in Ruby ... that should exercise the encoding aspect.

**Comment by ckumar1** [comment]
The issue was also definitely resolved in my test environment so we should be alright with the encoding. If we have any further issues with this, we can look into using the fast_blank implementation.

**Comment by jcheatham** [comment]
@ckumar1 - I just pulled a clean version of this repo and the test is failing:

```
$ bundle exec rake test
install -c tmp/x86_64-darwin13.0/rinku/2.1.5/rinku.bundle lib/rinku.bundle
cp tmp/x86_64-darwin13.0/rinku/2.1.5/rinku.bundle tmp/x86_64-darwin13.0/stage/lib/rinku.bundle
Run options: 

# Running tests:

[23/24] RedcarpetAutolinkTest#test_terminates_on_narrow_nbsp = 0.00 s                                   
  1) Failure:
RedcarpetAutolinkTest#test_terminates_on_narrow_nbsp [/Users/jcheatham/code/zendesk/rinku/test/autolink_test.rb:183]:
<"This is just a test. <a href=\"http://www.pokemon.com\">http://www.pokemon.com</a> "> expected but was
<"This is just a test. <a href=\"http://www.pokemon.com \">http://www.pokemon.com </a>">.

Finished tests in 0.021627s, 1109.7240 tests/s, 4115.2263 assertions/s.          
24 tests, 89 assertions, 1 failures, 0 errors, 0 skips

ruby -v: ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin13.0]
rake aborted!
Command failed with status (1): [ruby -I"lib" -rubygems -I"/opt/boxen/rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib" "/opt/boxen/rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/rake_test_loader.rb" "test/autolink_test.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

looks like we're not running travis on this.  Any thoughts?

**Pull request merged**